src/AutoRest.CSharp/Mgmt/Decorator/CodeModelExtension.cs

```diff
@@ -0,0 +1,179 @@
+                return;
+            var regexRawString = string.Join("|", RenameRules.Keys);
+            // we are building the regex that matches
+            // 1. it starts with a lower case letter or a digit which is considered as the end of its previous word
+            //    or it is at the beginning of this string
+            //    or we hit a word separator including \W, _, ., @, -, spaces
+            // 2. it either should match one of the candidates in the rename rules dictionary
+            // 3. it should followed by a upper case letter which is considered as the beginning of next word
+            //    or it is the end of this string
+            //    or we hit a word separator including \W, _, ., @, -, spaces
+            _regex = new Regex(@$"([\W|_|\.|@|-|\s|\$\da-z]|^)({regexRawString})([\W|_|\.|@|-|\s|\$A-Z]|$)");
+        }
+
+        public static void UpdateAcronyms(this IEnumerable<Schema> allSchemas, IReadOnlyDictionary<string, string> renameRules)
+        {
+            ApplyRenameRules(renameRules);
+            foreach (var schema in allSchemas)
+            {
+                TransformSchema(schema);
+            }
+        }
+
+        public static IEnumerable<Schema> GetAllSchemas(this CodeModel codeModel)
+        {
+            return codeModel.Schemas.Choices.Cast<Schema>()
+                .Concat(codeModel.Schemas.SealedChoices)
+                .Concat(codeModel.Schemas.Objects)
+                .Concat(codeModel.Schemas.Groups);
+        }
+
+        private static void TransformSchema(Schema schema)
+        {
+            switch (schema)
+            {
+                case ChoiceSchema:
+                case SealedChoiceSchema:
+                    TransformBasicSchema(schema);
+                    break;
+                case ObjectSchema objSchema: // GroupSchema inherits ObjectSchema, therefore we do not need to handle that
+                    TransformObjectSchema(objSchema);
+                    break;
+                default:
+                    throw new InvalidOperationException($"Unknown schema type {schema.GetType()}");
+            }
+        }
+
+        private static void TransformBasicSchema(Schema schema)
+        {
+            TransformLanguage(schema.Language);
+        }
+
+        private static void TransformLanguage(Languages languages)
+        {
+            var originalName = languages.Default.Name;
+            if (_cache.TryGetValue(originalName, out var result))
+            {
+                languages.Default.Name = result;
+                return;
+            }
+            result = originalName.EnsureNameCase();
+            languages.Default.Name = result;
+            _cache.TryAdd(originalName, result);
+        }
+
+        private static void TransformObjectSchema(ObjectSchema objSchema)
+        {
+            // transform the name of this schema
+            TransformBasicSchema(objSchema);
+            foreach (var property in objSchema.Properties)
```

Comment:
is this necessary since the loop that calls this is from allSchemas?

Reply:
it is necessary because this is changing the property's name not the type name of that property. Like in the following schema
```
public class Something {
    public PropertyType PropertyName {get; set;}
}
```
we are changing the property name, not the name of property type.